qa-core/src/internal-api/api/apis/DatasourcesAPI.ts

```diff
@@ -0,0 +1,92 @@
+import { Response } from "node-fetch"
+import { Datasource } from "@budibase/types"
+import { AddedDatasource } from "../../../types"
+import BudibaseInternalAPIClient from "../BudibaseInternalAPIClient"
+
+export default class DatasourcesAPI {
+  client: BudibaseInternalAPIClient
+
+  constructor(client: BudibaseInternalAPIClient) {
+    this.client = client
+  }
+
+  async getIntegrations(): Promise<[Response, any]> {
+    const [response, json] = await this.client.get(`/integrations`)
+    expect(response).toHaveStatusCode(200)
+    const integrationsCount = Object.keys(json).length
+    expect(integrationsCount).toBe(16)
+    return [response, json]
+  }
+
+  async getAll(): Promise<[Response, Datasource[]]> {
+    const [response, json] = await this.client.get(`/datasources`)
```

Comment:
I think every root api path should be it's own api, which helps with more consistent naming e.g.
```ts
config.api.datasources.getAll()
config.api.integrations.getAll()
config.api.queries...
config.api.permissions...
```
etc.